js/build_map.js

```diff
@@ -140,12 +142,18 @@ var categoryButtons = d3.select('#button-container')
   .on('click', function(d){
     prevCategory = activeCategory;
     activeCategory = d.toLowerCase(); // put this here so it only changes on click
-    updateCategory(activeCategory, action = 'click');
+    updateCategory(activeCategory, prevCategory.toLowerCase(), 
+    action = 'click');
```

Comment:
I'm wondering if we should be dividing the ideas in `updateCategory` into actions that need to happen when we preview a category and those that only happen when we click and commit to a category. For example, I believe we don't even need to be updating the category display (the circles) during calls to `on('click')` - by the time a mouse gets to clicking on a button, hasn't it already been hovering, and therefore aren't we already always viewing the category being clicked? So maybe `updateCategory` only needs to update the title, hash, category info, etc., like this: 

```
function updateCategory(category, prevCategory) {
      // update the globals about category view status
      prevCategory = activeCategory;
      activeCategory = d.toLowerCase();

  // update page info
  updateTitle(category);
  setHash('category', category);

  documentCategorySwitch(category, prevCategory, action='click');
}
  
function showCategory(category, prevCategory, action) {
  updatePies(category, prevCategory);
  documentCategorySwitch(category, prevCategory, action);
} 

function documentCategorySwitch (category, prevCategory, action) {
  if(updateCategoryTimer){
    clearTimeout(updateCategoryTimer);
  }
  updateCategoryTimer = setTimeout(function(){
     gtag('event', action + ' update category', {
  'event_category': 'figure',
  'event_label': category + '; view=' + activeView
     });
  }, updateCategoryDelay);
}
```

where `.on('click')` calls updateCategory whereas `.on('mouseover')` and `.on('mouseout')` call `showCategory`

I'm just drafting, not testing, this code, but it might be a start on finishing out task 1 - what do you think?

Reply:
Yeah, that sounds good

Reply:
yes, good point. maybe we make it two different gtag calls, one in updateCategory (without a timer) and one in showCategory (with a timer).